psd-web/app/controllers/investigations/test_results_controller.rb

```diff
@@ -4,4 +4,39 @@ def show
     authorize @investigation, :view_non_protected_details?
     @test_result = @investigation.test_results.find(params[:id]).decorate
   end
+
+  def edit
+    @investigation = Investigation.find_by(pretty_id: params[:investigation_pretty_id]).decorate
```

Comment:
```suggestion
    @investigation = Investigation.find_by!(pretty_id: params[:investigation_pretty_id]).decorate
```

Raises exception if not found.